packages/fulfillment/src/services/fulfillment-module-service.ts

```diff
@@ -152,14 +157,24 @@ export default class FulfillmentModuleService<
 
     if (context) {
       shippingOptions = shippingOptions.filter((shippingOption) => {
-        if (!shippingOption.rules?.length) {
+        if (!isPresent(shippingOption.conditions)) {
           return true
         }
 
-        return isContextValid(
-          context,
-          shippingOption.rules.map((r) => r)
-        )
+        const validate = ajv.compile({
```

Comment:
Of course we can create a tiny util that takes care of compiling and validating. We could also memoize the compilation for improved performance, but I don't think it will have significant impact anyway.